"panic: interface conversion" when reading an MP4 by accident

Repro is with `zq` commit `dba9ff1e`.

I happened to run `zq` from the wrong directory, such that it tried to read [this MP4 file](https://github.com/brimsec/zq/files/6032323/video.mp4.gz) as if it were a log:

```
$ zq -t 'count()' video.mp4.gz 
panic: interface conversion: zng.Type is *zng.TypeOfUint8, not *zng.TypeRecord

goroutine 1 [running]:
github.com/brimsec/zq/zng.NewVolatileRecordFromType(...)
	/Users/phil/work/zq/zng/recordval.go:109
github.com/brimsec/zq/zio/zngio.(*Reader).parseValue(0xc000375810, 0x0, 0x0, 0xc000410002, 0x0, 0x7fffe, 0x0, 0x1, 0x80000)
	/Users/phil/work/zq/zio/zngio/reader.go:493 +0x44f
github.com/brimsec/zq/zio/zngio.(*Reader).readValue(0xc000375810, 0x0, 0x0, 0x1, 0x80000, 0x0)
	/Users/phil/work/zq/zio/zngio/reader.go:202 +0x275
github.com/brimsec/zq/zio/zngio.(*Reader).readPayload(0xc000375810, 0x0, 0xc000375810, 0x1efe480, 0x1efe480, 0x0)
	/Users/phil/work/zq/zio/zngio/reader.go:149 +0x374
github.com/brimsec/zq/zio/zngio.(*Reader).ReadPayload(0xc000375810, 0x100fff8, 0x70, 0x18e2220, 0x18e3301)
	/Users/phil/work/zq/zio/zngio/reader.go:108 +0x38
github.com/brimsec/zq/zio/zngio.(*Reader).Read(0xc000375810, 0x18e2220, 0xc000375810, 0x2e)
	/Users/phil/work/zq/zio/zngio/reader.go:95 +0x2b
github.com/brimsec/zq/zio/detector.match(0x1a85b00, 0xc000375810, 0x191a109, 0x3, 0x80000, 0xa00000)
	/Users/phil/work/zq/zio/detector/reader.go:104 +0x38
github.com/brimsec/zq/zio/detector.NewReaderWithOpts(0x1a84640, 0xc000153080, 0xc000118708, 0x7ffeefbffa0e, 0xc, 0x191a5f2, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/phil/work/zq/zio/detector/reader.go:81 +0xfec
github.com/brimsec/zq/zio/detector.OpenFromNamedReadCloser(0xc000118708, 0x2f77998, 0xc000118710, 0x7ffeefbffa0e, 0xc, 0x191a5f2, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/phil/work/zq/zio/detector/file.go:74 +0x13e
github.com/brimsec/zq/zio/detector.OpenFileWithContext(0x1a95fa0, 0xc00011e008, 0xc000118708, 0x7ffeefbffa0e, 0xc, 0x191a5f2, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/phil/work/zq/zio/detector/file.go:44 +0x218
github.com/brimsec/zq/zio/detector.OpenFile(...)
	/Users/phil/work/zq/zio/detector/file.go:26
github.com/brimsec/zq/cli/inputflags.(*Flags).Open(0xc00011d9a0, 0xc000118708, 0xc000122030, 0x1, 0x1, 0x1a84f01, 0xc000380cf0, 0x0, 0x0, 0x203000, ...)
	/Users/phil/work/zq/cli/inputflags/flags.go:91 +0x198
main.(*Command).Run(0xc00011d980, 0xc000122020, 0x2, 0x2, 0x0, 0x0)
	/Users/phil/work/zq/cmd/zq/zq.go:159 +0x5b1
github.com/mccanne/charm.(*instance).run(0xc000111ee0, 0xc000122010, 0x3, 0x3, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x293
github.com/mccanne/charm.(*Spec).ExecRoot(0x1ef4400, 0xc000122010, 0x3, 0x3, 0xc00019ff78, 0x1007f65, 0xc0000a6058, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/phil/work/zq/cmd/zq/main.go:9 +0x85
```

Normally when I read garbage data, I'm accustomed to seeing the errors from the failed auto-detection attempts, e.g.:

```
$ zq -t 'count()' /etc/passwd
/etc/passwd: format detection error
	tzng: line 1: bad format
	zeek: line 11: bad types/fields definition in zeek header
	no json type config: ndjson detector skipped
	zjson: line 1: invalid character '#' looking for beginning of value
	zson: zson syntax error
	zng: zng type ID out of range
	parquet: auto-detection not supported
	zst: auto-detection not supported
```

--- a/zio/detector/ztests/fake-zng.yaml
+++ b/zio/detector/ztests/fake-zng.yaml
@@ -1,0 +1,10 @@
+script: zq -f tzng -
+
+inputs:
+  - name: stdin
+    data: !!binary |
+      AAAAHGZ0eXBtcDQyAAAAAWlzb21tcDQxbXA0MgAA
+
+outputs:
+  - name: stderr
+    regexp: .*non-record, top-level zng values are not supported

--- a/zio/zngio/reader.go
+++ b/zio/zngio/reader.go
@@ -482,7 +482,7 @@ func (r *Reader) parseValue(rec *zng.Record, id int, b []byte) (*zng.Record, err
 		return nil, zng.ErrTypeIDInvalid
 	}
 	if _, ok := zng.AliasedType(typ).(*zng.TypeRecord); !ok {
-		// An "id" of a top-level zng value not conformign with a
+		// An "id" of a top-level zng value not conforming with a
 		// record type is valid zng data but not supported by zq.
 		// This can also happen when trying to parse random non-zng
 		// data by the auto-dector.

--- a/zio/zngio/reader.go
+++ b/zio/zngio/reader.go
@@ -481,6 +481,14 @@ func (r *Reader) parseValue(rec *zng.Record, id int, b []byte) (*zng.Record, err
 	if err != nil {
 		return nil, zng.ErrTypeIDInvalid
 	}
+	if _, ok := zng.AliasedType(typ).(*zng.TypeRecord); !ok {
+		// An "id" of a top-level zng value not conformign with a
+		// record type is valid zng data but not supported by zq.
+		// This can also happen when trying to parse random non-zng
+		// data by the auto-dector.
+		return nil, errors.New("non-record, top-level zng values are not supported")
+	}
+
 	sharedType := r.mapper.Map(id)
 	if sharedType == nil {
 		var err error

